action.yml

```
@@ -0,0 +1,237 @@
+        fi
+
+        # Determine publishing conditions
+        PUBLISH_TO_GHCR="false"
+        PUBLISH_TO_GAR="false"
+
+        # GHCR: Publish on default branch push OR PR (if enabled)
+        if [[ "${{ inputs.ghcr }}" == "true" && (("${{ github.event_name }}" == "push" && "${{ github.ref_name }}" == "${{ github.event.repository.default_branch }}") || ("${{ github.event_name }}" == "pull_request" && "${{ inputs.publish_on_pr }}" == "true")) ]]; then
+          PUBLISH_TO_GHCR="true"
+        fi
+
+        # GAR: Publish on default branch push only
+        if [[ "${{ inputs.google_ar }}" == "true" && "${{ github.event_name }}" == "push" && "${{ github.ref_name }}" == "${{ github.event.repository.default_branch }}" ]]; then
+          PUBLISH_TO_GAR="true"
+        fi
+
+        echo "publish_to_ghcr=$PUBLISH_TO_GHCR" >> $GITHUB_OUTPUT
+        echo "publish_to_gar=$PUBLISH_TO_GAR" >> $GITHUB_OUTPUT
+        echo "Publishing to GHCR: $PUBLISH_TO_GHCR"
+        echo "Publishing to GAR: $PUBLISH_TO_GAR"
+
+
+    - name: Extract metadata for GHCR
+      id: meta-ghcr
+      if: steps.setup.outputs.publish_to_ghcr == 'true'
+      uses: docker/metadata-action@902fa8ec7d6ecbf8d84d538b9b233a880e428804 # v5.7.0
+      with:
+        images: ${{ steps.setup.outputs.ghcr_image_name }}
+        tags: |
+          # SHA for all events
+          type=sha,format=short
+          # Long SHA for all events
+          type=sha,format=long
+          # Nightly tags only for default branch pushes
+          type=raw,value=nightly,enable={{is_default_branch}}
+        labels: |
+            org.opencontainers.image.title=${{ github.event.repository.name }}
+            org.opencontainers.image.description=${{ github.event.repository.description }}
+            org.opencontainers.image.vendor="Sentry"
+            org.opencontainers.image.revision=${{ github.pull_request.head.sha || github.sha }}
+            org.opencontainers.image.version=${{ github.pull_request.head.sha || github.sha }}
+            org.opencontainers.image.source=${{ github.server_url }}/${{ github.repository }}
+            org.opencontainers.image.url=${{ github.server_url }}/${{ github.repository }}
+
+    - name: Login to GitHub Container Registry
+      if: steps.setup.outputs.publish_to_ghcr == 'true'
+      shell: bash
+      run: |
+        docker login ghcr.io -u "${{ github.actor }}" --password "${{ github.token }}"
+
+    - name: Build and push to GHCR
+      id: build-and-push-to-ghcr
+      if: steps.setup.outputs.publish_to_ghcr == 'true'
+      uses: docker/build-push-action@263435318d21b8e681c14492fe198d362a7d2c83 # v6.18.0
+      with:
+        context: ${{ inputs.build_context }}
+        file: ${{ inputs.dockerfile_path }}
+        platforms: linux/${{ inputs.platform }}
+        build-args: ${{ inputs.build_args }}
+        target: ${{ inputs.build_target }}
+        tags: ${{ steps.meta-ghcr.outputs.tags }}
+        labels: ${{ steps.meta-ghcr.outputs.labels }}
```

```suggestion
        labels: ${{ steps.meta-ghcr.outputs.labels }}
        annotations: ${{ steps.meta-ghcr.outputs.annotations }}
```

Include annotations too.